src/providers/vod/vod-provider.ts

```diff
@@ -127,8 +127,12 @@ export class VodProvider extends Provider {
     if (!match || !match[1]) {
       return; // captionAssetId not found;
     }
+    // as 'serveAsJson' action returns content of file it can't be included in multirequest that contains several responses.
+    // to prevent add in multirequest 'startWidgetSession' action for anonymous users doRequest method takes a session ks.
+    const {session} = this._player.config;
+    const ks = session?.isAnonymous ? this._player.config.session.ks : undefined;
     this._player.provider
-      .doRequest([{loader: CaptionLoader, params: {captionAssetId: match[1]}}])
+      .doRequest([{loader: CaptionLoader, params: {captionAssetId: match[1]}}], ks)
```

Comment:
@semarche-kaltura not sure where doRequest in defined (didn't find it in the player repo) but is it safe to send it an undefined parameter for ks?